saltyrtc/server/events.py

```diff
@@ -1,27 +1,67 @@
 import collections
 import enum
-from typing import List
+from typing import (
+    Callable,
+    Coroutine,
+    Dict,
+    List,
+    Optional,
+)
 
-try:
-    from collections.abc import Coroutine
-except ImportError:  # python 3.4
-    from backports_abc import Coroutine
+__all__ = (
+    'DisconnectedData',
+    'EventData',
+    'EventCallback',
+    'Event',
+    'EventRegistry',
+)
 
-__all__ = ('Event',)
+
+# Types
+DisconnectedData = int
+EventData = Optional[DisconnectedData]
+EventCallback = Callable[[Optional[str], EventData], Coroutine],
```

Comment:
Why is the first parameter optional? Don't we always pass it in?

Reply:
The path may not always be available.

Reply:
But you're correct that the first argument is missing which lead to this confusion...